app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -195,35 +228,11 @@ function ContainerTopologyCtrl($scope, $http, $interval, $location, topologyServ
   };
 
   this.getIcon = function getIcon(d) {
-    switch (d.item.kind) {
-      case 'Container':
-        return '\uF1B2'; // fa-cube
-      case "ContainerNode":
-        return '\uE621';  // pficon-container-node
-      case "ContainerRoute":
-        return '\uE625'; // pficon-route
-      case "ContainerService":
-        return '\uE61E'; // pficon-service
-      case "Vm":
-        return '\uE90f'; // pficon-virtual-machine
-      case "Host":
-        return '\uE600'; // pficon-screen
-      case "ContainerGroup":
-        return '\uF1B3'; // fa-cubes
-      case "ContainerReplicator":
-        return '\uE624'; // pficon-replicator
-      case "ContainerManager":
-        switch (d.item.display_kind) {
-          case "Kubernetes":
-            return '\uE627'; // pficon-kubernetes
-          case "Openshift":
-          case "OpenshiftEnterprise":
-            return '\uE626';  // pficon-openshift
-          case "Atomic":
-            return '\uE62c'; // vendor-atomic
-          case "AtomicEnterprise":
-            return '\uE62d'; //vendor-atomic-enterprise
-        }
+    switch(d.item.kind) {
+      case 'ContainerManager':
+        return icons[d.item.display_kind];
```

Comment:
@martinpovolny @himdel I'm unclear how do you propose to solve it differently, and what are the data tags you are talking about. I'd be happy to see a concrete example.
Please note that:
a. in topology which is a one big svg, svg icons and text icons are treated differently (one is text and the other is img) 
b. I tried to refrain from attaching an icon information to each entity in the json data, because then it would blow the json to a much bigger size with same repeated data (so for example if there are 200 entities of type pod, and all pods have the same icon, I wanted to avoid the json to have 200 more entries with same icon information)